src/System.Buffers.Primitives/System/Internal/ManagedBufferPool.cs

```diff
@@ -22,7 +22,8 @@ public static ManagedBufferPool Shared
 
         public override OwnedBuffer<byte> Rent(int minimumBufferSize)
         {
-            return new ArrayPoolBuffer(minimumBufferSize);
+            var buffer = new ArrayPoolBuffer(minimumBufferSize);
+            return buffer;
```

Comment:
I was increment the refcount in Rent. Then I removed the increment. I could rollback this change now, but it makes it easier to experiment with refcounting. If you feel like we should roll back, I will do.